packages/storage-plugin/tests/storage.plugin.spec.ts

```diff
@@ -462,4 +478,99 @@ describe('NgxsStoragePlugin', () => {
       });
     });
   });
+
+  describe('Custom serialization', () => {
+    it('should alter object before serialization.', () => {
+      // Arrange
+      localStorage.setItem(DEFAULT_STATE_KEY, JSON.stringify({ counter: { count: 100 } }));
+
+      const serialization = new SerializationStrategy([
+        {
+          onBeforeSerialize: obj => {
+            return {
+              counter: {
+                count: obj.counter.count * 2
+              }
+            };
+          }
+        }
+      ]);
+
+      // Act
+      TestBed.configureTestingModule({
+        imports: [
+          NgxsModule.forRoot([CounterState]),
+          NgxsStoragePluginModule.forRoot({
+            serialize: (obj, key) => serialization.serialize(obj, key)
+          })
+        ]
+      });
+
+      const store: Store = TestBed.get(Store);
+
+      store.dispatch(new Increment());
+
+      const state: CounterStateModel = store.selectSnapshot(CounterState);
+
+      // Assert
+      expect(state.count).toBe(101);
+      expect(localStorage.getItem(DEFAULT_STATE_KEY)).toBe(
+        JSON.stringify({ counter: { count: 202 } })
+      );
+    });
+
+    it('should alter state and return concrete type after deserialization.', () => {
+      // Arrange
+      localStorage.setItem('counterInfo', JSON.stringify({ count: 100 }));
+
+      const serialization = new SerializationStrategy([
+        {
+          key: 'counterInfo',
+          onAfterDeserialize: obj => new CounterInfoStateModel(obj.count)
+        }
+      ]);
+
+      // Act
+      TestBed.configureTestingModule({
+        imports: [
+          NgxsModule.forRoot([CounterInfoState]),
+          NgxsStoragePluginModule.forRoot({
+            key: 'counterInfo',
+            deserialize: (obj, key) => serialization.deserialize(obj, key)
+          })
+        ]
+      });
+
+      const store: Store = TestBed.get(Store);
+      const state: CounterInfoStateModel = store.selectSnapshot(CounterInfoState);
+
+      // Assert
+      expect(state).toBeInstanceOf(CounterInfoStateModel);
+      expect(state.count).toBe(100);
+    });
+
+    it('Should not introduce any breaking changes.', () => {
+      // Arrange
+      localStorage.setItem('counter', JSON.stringify({ count: 100 }));
+
+      const deserialize = (obj: any): any => JSON.parse(obj);
+
+      // Act
+      TestBed.configureTestingModule({
+        imports: [
+          NgxsModule.forRoot([CounterState]),
+          NgxsStoragePluginModule.forRoot({
+            key: 'counter',
+            deserialize: deserialize
```

Comment:
Please change this line as follows:
```suggestion
            deserialize: JSON.parse
```
If this change works then I would be confident that this is not a breaking change.